Sanguine Soul/sanguinesoul.asl

```diff
@@ -54,36 +56,40 @@ init
 update
 {
     // Deref useful FNames here
-    IDictionary<string, object> currdict = current;
-    foreach (var fname in new List<string>(currdict.Keys))
+    IDictionary<string, object> currentLookup = current;
+    foreach (var key in new List<string>(currentLookup.Keys))
```

Comment:
This was necessary, since if we directly iterate over the dictionary we get concurrency issues (can't mutate the dictionary while iterating over it)